packages/relay/src/index.ts

```diff
@@ -249,7 +250,17 @@ export function wakuGossipSub(
   return (components: GossipSubComponents) => {
     init = {
       ...init,
-      msgIdFn: ({ data }) => sha256(data),
+      msgIdFn: ({ topic, data }) => {
+        try {
+          const protoMessage = message.fromBytesToProtoMessage(data);
+          return messageHash(topic, protoMessage);
```

Comment:
Problem with this implementation is that in case `meta` is missing hash is taken based on this `concat([pubsubTopicBytes, message.payload, contentTopicBytes])` 

It leads to inability of a user to send a message to the a pubSub/contentTopic in case someone already published a message with the same text. 

Second problem of using `messageHash` as `msgIdFn` is that it won't be consistent with hashing in `nwaku`/`go-waku` as I mentioned [here](https://github.com/waku-org/js-waku/issues/1179#issuecomment-1532242636)